Add allgather_into_tensor to CommTensor

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #90681
* #90569
* __->__ #90565
* #90564
* #90512

This PR adds _all_gather_base_ to CommTensor to support allgather_base

wanchaol committed:
Add allgather_into_tensor to CommTensor

This PR adds _all_gather_base_ to CommTensor to support allgather_base

[ghstack-poisoned]

wanchaol committed:
Update on "Add allgather_into_tensor to CommTensor"


This PR adds _all_gather_base_ to CommTensor to support allgather_base

[ghstack-poisoned]

mrshenli commented on torch/distributed/_spmd/comm_tensor.py:
```diff
@@ -95,6 +95,7 @@ class CommTensor(torch.Tensor):
         "broadcast_",
         "reduce_scatter_",
         "scatter_",
+        "_allgather_base_",
```
nit: `_` is before lower case letters in ASCII order

mrshenli reviewed (approved):
Stamp to unblock

wanchaol committed:
Update on "Add allgather_into_tensor to CommTensor"


This PR adds _all_gather_base_ to CommTensor to support allgather_base

[ghstack-poisoned]

wanchaol committed:
Update on "Add allgather_into_tensor to CommTensor"


This PR adds _all_gather_base_ to CommTensor to support allgather_base

[ghstack-poisoned]

wanchaol committed:
Update on "Add allgather_into_tensor to CommTensor"


This PR adds _all_gather_base_ to CommTensor to support allgather_base

[ghstack-poisoned]

wanchaol committed:
Update on "Add allgather_into_tensor to CommTensor"


This PR adds _all_gather_base_ to CommTensor to support allgather_base

[ghstack-poisoned]
